ui/apps/platform/src/Containers/Collections/CollectionResults.tsx

```diff
@@ -54,104 +63,146 @@ function CollectionResults({ dryRunConfig }: CollectionResultsProps) {
         closeSelect();
     }
 
-    const fetchDryRun = useCallback((currConfig, currPage, currFilter, currEntity) => {
-        fetchMatchingDeployments(currConfig, currPage, currFilter, currEntity)
-            .then((results) => {
-                setIsEndOfResults(results.length < 10);
-                setDeployments((current) => (currPage === 0 ? results : [...current, ...results]));
-            })
-            .catch(() => {
-                // TODO: indicate results not loading properly?
-            });
-    }, []);
+    const fetchDryRun = useCallback(
+        (currConfig, currPage, currFilter, currEntity) => {
+            fetchMatchingDeployments(currConfig, currPage, currFilter, currEntity)
+                .then((results) => {
+                    setIsEndOfResults(results.length < 10);
+                    setDeployments((current) =>
+                        currPage === 0 ? results : [...current, ...results]
+                    );
+                })
+                .catch((err) => {
+                    setSaveError(parseSaveError(err));
+                });
+        },
+        [setSaveError]
+    );
 
     const fetchDryRunDebounced = useMemo(() => debounce(fetchDryRun, 800), [fetchDryRun]);
 
     useEffect(() => {
+        if (saveError) {
+            setDeployments([]);
```

Comment:
Flush old data on error to prevent the display of incorrect deployments the next time the rules correctly validate.